docker/docker-entrypoint.sh

```diff
@@ -1,56 +1,90 @@
-#!/bin/bash -e
+#!/usr/bin/env bash
+
+set -e -u -o pipefail
 
 function alldone() {
-    # send SIGINT to ziti-tunnel to trigger a cleanup of iptables mangle rules
-    kill -INT $ZITI_EDGE_TUNNEL_PID
-    # let entrypoint script exit after ziti-tunnel PID
-    wait $ZITI_EDGE_TUNNEL_PID
+    # if successfully sent to background then send SIGINT to trigger a cleanup
+    # of iptables mangle rules and loopback assignments
+    [[ "${ZITI_EDGE_TUNNEL_PID:-}" =~ ^[0-9]+$ ]] && {
+        kill -INT "$ZITI_EDGE_TUNNEL_PID"
+        # let entrypoint script exit after ziti-tunnel PID
+        wait "$ZITI_EDGE_TUNNEL_PID"
+    }
 }
 trap alldone exit
 
-# Ensure that ziti-tunnel's identity is stored on a volume
+# Ensure that ziti-edge-tunnel's identity is stored on a volume
 # so we don't throw away the one-time enrollment token
 
-persisted_dir="/ziti-edge-tunnel"
-wc_lines=$(df / "${persisted_dir}" 2> /dev/null | uniq | wc -l)
-if [ "${wc_lines}" != "3" ]; then
-    echo "ERROR: please run this image with ${persisted_dir} mounted on a volume"
+IDENTITIES_DIR="/ziti-edge-tunnel"
+if ! mountpoint "${IDENTITIES_DIR}" &>/dev/null; then
+    echo "ERROR: please run this image with a volume mounted on ${IDENTITIES_DIR}" >&2
     exit 1
 fi
 
-# try to figure out the client name if it wasn't provided
-if [ -z "${NF_REG_NAME}" ]; then
-    if [ -n "${IOTEDGE_DEVICEID}" ]; then
-        echo "setting NF_REG_NAME to \${IOTEDGE_DEVICEID} (${IOTEDGE_DEVICEID})"
+# TODO: What is the origin story of IOTEDGE_DEVICEID and can it be decommissioned? -Ken
+if [[ -z "${NF_REG_NAME:-}" ]]; then
+    if [[ -n "${IOTEDGE_DEVICEID:-}" ]]; then
+        echo "INFO: setting NF_REG_NAME to \${IOTEDGE_DEVICEID} (${IOTEDGE_DEVICEID})"
         NF_REG_NAME="${IOTEDGE_DEVICEID}"
     fi
 fi
-if [ -z "${NF_REG_NAME}" ]; then
```

Comment:
I think this can only be removed if the IOTEDGE_DEVICEID logic is removed. This prevents falling through without an identity name.

Reply:
The net effect of the changes is currently:

```bash
# pseudo-bash
if NF_REG_NAME; then
    identity is NF_REG_NAME
elif IOTEDGE_DEVICEID; then
    identity is IOTEDGE_DEVICEID
else
    false
fi
```
Do you think that'll work with the way you remember the Azure env var working?